tests/integration_tests/performance/test_vsock_throughput.py

```diff
@@ -350,7 +357,10 @@ def iperf_workload(context):
     except core.CoreException as err:
         handle_failure(file_dumper, err)
 
-    dump_test_result(file_dumper, result)
+    # This will dump the things we told them to dump
+    # in the pytest provided parameters.
+    if file_dumper:
```

Comment:
Proposal: as noted in a previous comment, the `file_dumper` parameter could be an instance that implements an early exit for the `dump()` call.

This way you can avoid checking for `if file_dumper`.

Reply:
Done.